### ISSUE TITLE
V3: Chapter navigation not functional in Chrome (Mac OS X)

### ISSUE THREAD

**Issue by stenskjaer** [podlove/podlove-web-player#284]
When trying to change the chapter playing in Chrome, the recording just starts from the beginning.
Example: http://188.166.71.115/listen/12-2
In Firefox there is no problem.
Can you recognize (replicate, maybe even) that problem?

**Comment by katrinleinweber** [comment]
I see the same problem with Chrome 46.0.2490.86 64bit on Mac OS X 10.11 on your website. On [my website](http://konscience.de/) it works, though.

**Comment by plutonik-a** [comment]
In Safari / OSX, chapters don't work at all.
Without knowing what's causing the problem, my first clue is `jquery.min.js` was included twice and your side navigation is linked with the chapter marks. 
Could this possibly interfere with the player script?
What happens, if you deactivate the side navigation for a try?

**Comment by stenskjaer** [comment]
To follow up on this, I have compared the console logs from Chrome and Firefox...

```
Chapter clickHandler setCurrentChapter to 3
Chapters getActiveChapter Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…}code: "VII. The king in Kotorr, sung"duration: "16:22"element: m.fn.init[1]end: 1454song: "15"start: 472startTime: "07:52"title: "VII. The king in Kotorr, sung"__proto__: Object
Chapters setCurrentChapter to 3
Chapters getActiveChapter Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…}
Chapters #playCurrentChapter start 472
Timeline setTime time 472
Timeline update undefined
Timeline setBufferedTime 0.235102
Timeline currentTime 472
Chapters getActiveChapter Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…}
Chapters update d {timeline: s, duration: 3932.577959, chapterlinks: false, currentChapter: 3, chapters: Array[9]…} Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…} 472
Chapters update already set 3
SaveTime update 472
ProgressBar setHandlePosition time 472 newLeftOffset 12.002304974521676%
canplay setTime playerState 4
Chapters #playCurrentChapter currentTime 472
Timeline update m.Event {originalEvent: Event, type: "timeupdate", timeStamp: 1448304457122, jQuery111306486049727536738: true, which: undefined…}
Timeline setBufferedTime 0.235102
Timeline currentTime 0
Chapters getActiveChapter Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…}
Chapters update d {timeline: s, duration: 3932.577959, chapterlinks: false, currentChapter: 3, chapters: Array[9]…} Object {start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454…} 0
Chapters getActiveChapter Object {start: 0, title: "Introduction", code: "Introduction", end: 0, duration: "00:00"…}
Chapters setCurrentChapter to 0
Chapters getActiveChapter Object {start: 0, title: "Conversation", code: "Conversation", end: 50, duration: "0:50"…}
Chapters setCurrentChapter to 1
SaveTime update 0
ProgressBar setHandlePosition time 0 newLeftOffset 0%
```

Firefox:
```
Chapter clickHandler setCurrentChapter to 3 
Chapters getActiveChapter Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object } 
Chapters setCurrentChapter to 3 
Chapters getActiveChapter Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object } 
Chapters #playCurrentChapter start 472
Timeline setTime time 472
Timeline update undefined
Timeline setBufferedTime 3932.634965
Timeline currentTime 472 
Chapters getActiveChapter Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object } 
Chapters update Object { timeline: Object, duration: 3932.634965, chapterlinks: false, currentChapter: 3, chapters: Array[9], data: Array[9], tab: Object, update: l() } Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object } 472 
Chapters update already set 3
SaveTime update 472 
ProgressBar setHandlePosition time 472 newLeftOffset 12.002130993614863% 
canplay setTime playerState 4 
Chapters #playCurrentChapter currentTime 472 
Timeline update Object { originalEvent: timeupdate, type: "timeupdate", isDefaultPrevented: ba(), timeStamp: 1448304704317266, jQuery11130055485279562808354: true, which: undefined, view: undefined, target: <audio>, shiftKey: undefined, relatedTarget: undefined, 10 mere… } 
Timeline setBufferedTime 3932.634965
Timeline currentTime 472 
Chapters getActiveChapter Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object }
Chapters update Object { timeline: Object, duration: 3932.634965, chapterlinks: false, currentChapter: 3, chapters: Array[9], data: Array[9], tab: Object, update: l() } Object { start: 472, title: "VII. The king in Kotorr, sung", song: "15", code: "VII. The king in Kotorr, sung", end: 1454, duration: "16:22", startTime: "07:52", element: Object } 472
Chapters update already set 3
SaveTime update 472
ProgressBar setHandlePosition time 472 newLeftOffset 12.002130993614863% 
```

A comparison of the two show that when when the `currentTime` it is set to 0. My guess is that is has something to do with the value `setBufferedTime`, which is almost 0 in chrome and just below 4000 in Firefox, which I think means that it is fully bufferede (length of this track is 01:05:32).
I can also see that the recording does not buffer at all in Chrome.
Might this be the explanation?

**Comment by line-o** [comment]
Interesting find @stenskjaer will look into this. Chrome might have changed the output of the buffering or this might be a server side problem.
By the way, make sure you use the very latest version of the player, as it is still not a (fully) stable release.

**Comment by line-o** [comment]
@stenskjaer your server behaves a little weird when attempting to download the mp3 in question. Safari needs to hand it over to some "plug-in" that handles the load. And via curl it takes *ages* to acquire that relatively small 45mb file and it does not even finish. Here is the verbose log of that attempt.
```
*   Trying 188.166.71.115...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* Connected to 188.166.71.115 (188.166.71.115) port 80 (#0)
> GET /static/recordings/12-2.mp3 HTTP/1.1
> Host: 188.166.71.115
> User-Agent: curl/7.43.0
> Accept: */*
>
  0     0    0     0    0     0      0      0 --:--:--  0:00:13 --:--:--     0< HTTP/1.1 200 OK
< Server: nginx/1.4.6 (Ubuntu)
< Date: Mon, 23 Nov 2015 23:48:29 GMT
< Content-Type: audio/mpeg
< Content-Length: 47191507
< Connection: keep-alive
< Last-Modified: Fri, 18 Sep 2015 13:53:53 GMT
< Cache-Control: public, max-age=43200
< Expires: Tue, 24 Nov 2015 11:48:29 GMT
< ETag: "flask-1442584433.57-47191507-2741768403"
<
{ [16049 bytes data]
  6 45.0M    6 3104k    0     0   6039      0  2:10:14  0:08:46  2:01:28  7467* transfer closed with 44012536 bytes remaining to read
* Closing connection 0
curl: (18) transfer closed with 44012536 bytes remaining to read
```

**Comment by line-o** [comment]
One other thing that always bugged me was that duration could not be read from the file.
Could you recompile that mp3 to a mp4 with headers at the beginning for testing?
